docs/source/using_yosys/more_scripting/data_flow_tracking.rst

```diff
@@ -0,0 +1,71 @@
+Dataflow tracking
+-------------------
+Yosys can be used to answer questions such as "can this signal affect this other signal?" via its *dataflow tracking* support.
+For this, four special cells, ``$get_tag``, ``$set_tag``, ``$overwrite_tag`` and ``$original_tag`` are inserted into the design and then the ``dft_tag`` is run, which converts these cells into ordinary logic.
+Typically, one would then use ``sby`` to prove assertions involving these cells.
+
+Ordinarily in Yosys, the state of a bit is simply ``0`` or ``1`` (or one of the special values, ``z`` and ``x``).
+During dataflow tracking they are augmented with a set of tags.
+For example, the state of a bit could be ``0`` and the set of tags ``"KEY"`` and ``"OVERFLOW"``.
+
+In addition to their usual operations on the logical bits, Yosys operations must now also process the status of the tags.
+For this, tags are simply *forwarded* or *propagated* (i.e. copied) from inputs to outputs, according to the following general rule:
+
+   A tag is forwarded from an input to an output if the input can affect the output, for that particular state of all other inputs.
+
+For example, XOR, AND and OR cells propagate tags as follows:
+
+#. XOR simply forwards all tags from its inputs to its output, because inputs to XOR can always affect the output.
+#. AND forwards tags only if the other input is ``1``. Because if one input is ``0``, the other input can never affect the output.
+#. Similarly, OR forwards tags only if the other input is ``0``.
+
+There are two exceptions to this rule:
+#. In general, propagation is only determined approximately.
```

Comment:
There needs to be an empty line before lists (the missing line breaks the formatting and issues warnings, and CI is failing with warnings-as-errors).
```suggestion
There are two exceptions to this rule:

#. In general, propagation is only determined approximately.
```